torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -640,9 +642,28 @@ void ProcessGroupNCCL::WorkNCCL::handleException(
 }
 
 void ProcessGroupNCCL::WorkNCCL::synchronize() {
-  // Call Synchronize without a timeout. We use this method to avoid adding a
-  // timeout argument to the public synchronize API.
```

Comment:
nit: can we add back some documentation on the behavior of this function?

Q: is this method being called at any other places? Just in case the `if (barrierTensor_.defined())` block (moved here) would change their expectation.

Reply:
"if (barrierTensor_.defined()) " block was actually already here in this API before (within synchronizeInternal). Since We removed synchronizeInternal, so we pop up the block here